Feature/object validation (#10)
CitrineInformatics/taurus-documentation: docs/philosophy.md

However, uncertainty quantification can be challenging and is often impossible in hindsight.
Therefore, `nominal` values are permitted to express "It was nominally X, but we don't know how certain we are about it."

#### Material history is chronological

> Starting from a given material, one can trace back the process that created the material, 

Author: ```suggestion
For any given physical object, one can trace back the series of processes that generated it and the ingredient materials that went into those processes. 
```

> the ingredients that went into the process, the materials that those ingredients were derived from, and so on.

Author: ```suggestion
```

> This history is chronological: for example, the ingredients that are involved in a process must exist *before* the process occurs.

Author: ```suggestion
We can represent this chronological history as a directed acyclic graph where each material links to the process that created it and each process links to the materials that went into it.
Each of those input materials must exist *before* the process occurs.
```

> As a consequence, there can be no cycles in a material history.

Reviewer: I can't figure out how to do multiline-suggestions, and spawning a PR seems a bit much.  I'd propose this reads:

For any physical object, one can trace back the series of processes that generated it and the ingredient materials that went into those processes.  
Formally, such a chronology represents a directed acyclic graph, where causal structure leads naturally to a topological ordering for nodes.  
Following this formalism and the common-sense idea that time moves in one direction only, a material history cannot contain a cycle, i.e., following references in the order material->process->ingredient->material will never see the same node twice.

> As a consequence, there can be no cycles in a material history.

Author: ```suggestion
As a consequence there can be no cycles in a well-formed material history, as that would imply a material being created and then traveling back in time to be used in one of the processes that were involved in creating it.
```

However, uncertainty quantification can be challenging and is often impossible in hindsight.
Therefore, `nominal` values are permitted to express "It was nominally X, but we don't know how certain we are about it."

#### Material history is chronological

For any given physical object, one can trace back the series of processes that generated it and the ingredient materials that went into those processes. 

> We can represent this chronological history as a directed acyclic graph where each material links to the process that created it and each process links to the materials that went into it.

Author: ```suggestion
We represent this chronological history as a directed acyclic graph where each material links to the process that created it and each process links to the materials that went into it.
```
